### PR TITLE
Replace `OnceLock` + `fn` with `LazyLock`

### PR DESCRIPTION
**Pull request by hinto-janai**
### What
Closes https://github.com/Cuprate/cuprate/issues/236.

This replaces the `OnceLock` + `fn`s that were used in a `LazyLock` type of way for the entire codebase. There's other `OnceLock`s remaining but they are used differently.

### CODE REVIEW

File: consensus/rules/src/decomposed_amount.rs [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-///
-static DECOMPOSED_AMOUNTS: OnceLock<[u64; 172]> = OnceLock::new();
+use std::sync::LazyLock;
 
 #[rustfmt::skip]
-pub fn decomposed_amounts() -> &'static [u64; 172] {
-    DECOMPOSED_AMOUNTS.get_or_init(|| {
-        [
+/// Decomposed amount table.
+pub static DECOMPOSED_AMOUNTS: LazyLock<[u64; 172]> = LazyLock::new(|| {
```
Comment (Boog900) [inline consensus/rules/src/decomposed_amount.rs:5]: Can we remove the `LazyLock` here, keep it a static though. I don't know why I made this a `OnceLock`, the data is still in the binary.

**Review by Boog900 — CHANGES_REQUESTED** [review]

File: consensus/rules/src/decomposed_amount.rs (hunk 1 above)
Comment (SyntheticBird45) [reply consensus/rules/src/decomposed_amount.rs:5]: I must say I'm confused on why this isn't a `const`

Comment (hinto-janai) [reply consensus/rules/src/decomposed_amount.rs:5]: `static` is a ref to 1 memory address containing the data, `const` inlines the data everywhere it gets used.

I assume @Boog900 doesn't want it inlined in the binary everywhere it gets used.

Comment (SyntheticBird45) [reply consensus/rules/src/decomposed_amount.rs]: I was confused, I thought const weren't inlined. Thanks for the heads-up

**Review by Boog900 — APPROVED** [review]

**Pull request merged**